src/fmu/ensemble/ensemble.py

```diff
@@ -234,21 +253,37 @@ def add_realizations(self, paths, realidxregexp=None, autodiscovery=True):
             globbedpaths = glob.glob(paths)
 
         count = 0
-        for realdir in globbedpaths:
-            realization = ScratchRealization(
-                realdir, realidxregexp=realidxregexp, autodiscovery=autodiscovery
-            )
-            if realization.index is None:
-                logger.critical(
-                    "Could not determine realization index " + "for path " + realdir
+        if USE_CONCURRENT:
+            with concurrent.futures.ProcessPoolExecutor() as executor:
+                for realization in executor.map(self.InitScratchRealization, globbedpaths):
```

Comment:
thinking loudly: possible to do `self.__init__` or something here? (to avoid the wrapper)

Reply:
No sure I fully get what you mean - elaborate?

Reply:
Avoiding a wrapper is what I mean, but I don't know how to achieve it..

Reply:
I think this is a standard way of interacting the `ProcessPoolExecutor()`.